File: Go/jobspls/algorithms/unpaired_occur.go
package algorithms

func unpaired_value[T comparable](haystack []T) (bool, T) {
	occur := make(map[T]int, len(haystack))
	for _, v := range haystack {
		occur[v] += 1
	}

	for key, count := range occur {
		if count%2 != 0 {
			return true, key
		}
	}

	return false, *new(T)
}

File: Java/jobspls/algorithms/UnpairedOccur.java
package jobspls.algorithms;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class UnpairedOccur {
    public static <T> Optional<T> unpaired_occur(T[] haystack) {
        var occur = new HashMap<T, Integer>(haystack.length);
        for (var v : haystack) {
            var count = occur.getOrDefault(v, 0);
            occur.put(v, count + 1);
        }

        return occur.entrySet()
                    .stream()
                    .filter(e -> e.getValue() % 2 != 0)
                    .map(Map.Entry::getKey)
                    .findFirst();
    }
}
